internal/bzlmod/semver.bzl

```diff
@@ -34,6 +34,8 @@ def _identifier_to_comparable(ident, *, numeric_only):
         # 11.4.3:
         # "Numeric identifiers always have lower precedence than non-numeric identifiers."
         return (_COMPARES_LOWEST_SENTINEL, int(ident))
+    elif ident == _COMPARES_HIGHEST_SENTINEL:
```

Comment:
Is this fixing a bug? I don't see right away hope it's related to the rest of the PR

Reply:
This PR gives _replaced go modules_ the version of `_COMPARES_HIGHEST_SENTINEL` so that we get the expected resolution behavior. When implementing that, I ran into this gap and had to close it.

I am of-course open to exploring an alternative approach, but this felt like the right one to me.